sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -0,0 +1,595 @@
+      .add("datasource", ValueType.STRING)
+      .add("created_time", ValueType.STRING)
+      .add("queue_insertion_time", ValueType.STRING)
+      .add("status", ValueType.STRING)
+      .add("runner_status", ValueType.STRING)
+      .add("duration", ValueType.LONG)
+      .add("location", ValueType.STRING)
+      .add("host", ValueType.STRING)
+      .add("plaintext_port", ValueType.STRING)
+      .add("tls_port", ValueType.STRING)
+      .add("error_msg", ValueType.STRING)
+      .build();
+
+  private final Map<String, Table> tableMap;
+
+  static {
+    SEGMENT_SERVERS_TABLE_SIZE = SERVERSEGMENTS_SIGNATURE.getRowOrder().size();
+  }
+
+  @Inject
+  public SystemSchema(
+      final DruidSchema druidSchema,
+      final TimelineServerView serverView,
+      final AuthorizerMapper authorizerMapper,
+      final @Coordinator DruidLeaderClient coordinatorDruidLeaderClient,
+      final @IndexingService DruidLeaderClient overlordDruidLeaderClient,
+      final ObjectMapper jsonMapper
+  )
+  {
+    Preconditions.checkNotNull(serverView, "serverView");
+    BytesAccumulatingResponseHandler responseHandler = new BytesAccumulatingResponseHandler();
+    this.tableMap = ImmutableMap.of(
+        SEGMENTS_TABLE, new SegmentsTable(druidSchema, coordinatorDruidLeaderClient, jsonMapper, responseHandler),
+        SERVERS_TABLE, new ServersTable(serverView),
+        SEGMENT_SERVERS_TABLE, new ServerSegmentsTable(serverView),
+        TASKS_TABLE, new TasksTable(overlordDruidLeaderClient, jsonMapper, responseHandler)
+    );
+  }
+
+  @Override
+  public Map<String, Table> getTableMap()
+  {
+    return tableMap;
+  }
+
+  static class SegmentsTable extends AbstractTable implements ScannableTable
+  {
+    private final DruidSchema druidSchema;
+    private final DruidLeaderClient druidLeaderClient;
+    private final ObjectMapper jsonMapper;
+    private final BytesAccumulatingResponseHandler responseHandler;
+
+    public SegmentsTable(
+        DruidSchema druidSchemna,
+        DruidLeaderClient druidLeaderClient,
+        ObjectMapper jsonMapper,
+        BytesAccumulatingResponseHandler responseHandler
+    )
+    {
+      this.druidSchema = druidSchemna;
+      this.druidLeaderClient = druidLeaderClient;
+      this.jsonMapper = jsonMapper;
+      this.responseHandler = responseHandler;
+    }
+
+    @Override
+    public RelDataType getRowType(RelDataTypeFactory typeFactory)
+    {
+      return SEGMENTS_SIGNATURE.getRelDataType(typeFactory);
+    }
+
+    @Override
+    public TableType getJdbcTableType()
+    {
+      return TableType.SYSTEM_TABLE;
+    }
+
+    @Override
+    public Enumerable<Object[]> scan(DataContext root)
+    {
+
+      //get available segments from druidSchema
+      Map<String, ConcurrentSkipListMap<DataSegment, SegmentMetadataHolder>> getSegmentMetadataInfo = druidSchema.getSegmentMetadataInfo();
+      final Map<DataSegment, SegmentMetadataHolder> availableSegmentMetadata = new HashMap<>();
```

Comment:
What does `availableSegment` mean?

Reply:
Well, if they don't have the clear meaning, I think it's better to call them just `segmentsMetadataFromDruidSchema` and `dataSegmentsFromCoordinator`. The name is quite long, but more clear.

Reply:
Would you please check this?